File: muxer-commons/src/main/java/pt/um/tf/taskmux/commons/error/NoAssignableTasksException.java
package pt.um.tf.taskmux.commons.error;

import io.atomix.catalyst.buffer.BufferInput;
import io.atomix.catalyst.buffer.BufferOutput;
import io.atomix.catalyst.serializer.CatalystSerializable;
import io.atomix.catalyst.serializer.Serializer;

public class NoAssignableTasksException extends Exception implements CatalystSerializable {
    @Override
    public String getMessage() {
        return "No Tasks present to assign right now";
    }

    @Override
    public void writeObject(BufferOutput<?> buffer, Serializer serializer) {

    }

    @Override
    public void readObject(BufferInput<?> buffer, Serializer serializer) {

    }
}

File: muxer-commons/src/main/kotlin/pt/um/tf/taskmux/commons/error/NoAssignableTasksException.kt
package pt.um.tf.taskmux.commons.error

import io.atomix.catalyst.buffer.BufferInput
import io.atomix.catalyst.buffer.BufferOutput
import io.atomix.catalyst.serializer.CatalystSerializable
import io.atomix.catalyst.serializer.Serializer

class NoAssignableTasksException : Exception(), CatalystSerializable {
    override val message: String?
        get() = "No Tasks present to assign right now"

    override fun writeObject(buffer: BufferOutput<*>?, serializer: Serializer?) {}

    override fun readObject(buffer: BufferInput<*>?, serializer: Serializer?) {}

}
